### PR TITLE
Redesign website

### LINKED ISSUE #6: Animate list of covers
I was expecting the list of covers to be shown like a slide show, but at the moment only the first cover image is shown for " 3D UNet Platynereis Cell Segmentation (Probabilities)"[name is also too long atm]
This is particularly unfortunate, as the first one shows only raw data.
image

For "3D UNet Arabidopsis Ovules" this is the case on the preview windows, but there is another problem: the card does not show any cover image at all.
image
image

### LINKED ISSUE #22: Support structured tags and searching
Please comment below for the categories we want to use to structure the tags.
  * Modality (type of microscope)
  * Organism (animal, tissue type etc.)
  * Software (imagej, ilastik, imjoy, ...)
  * Model Architecture (UNet, ResNet, ...)
  * Task (Segmentation, Classification, Denoising, ...)

### LINKED ISSUE #23: Support free text searching

### PR DESCRIPTION
**Pull request by oeway**
This PR replace the static pages into a full webpack based VueJS project, also uses github actions for CI, redesign the interface with Buefy/Bulma css framework.

* I reorganised the project so we can use npm build system and CI to generate the site
* Make the model cards more general such that we can also include datasets and applications, also notebooks can be added later
* The search feature is much more improved, now we support also free text, there is also tag groups as suggested by Anna
* Added badges, for example the CO2 footprint badge. Later, we can add also certificate for model reproducibility, tests passed etc.
* Each model is now associated with an anonymous animal to make it more like a zoo ;)
* Multiple cover images are shown as carousel 
* Use github style to show documentation
* Improved dialog window that can be maximized and dragged
* Support both `list` and `card` display mode

Fixes #6
Fixes #22 
Fixes #23

### CODE REVIEW

File: src/views/Home.vue [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+          <h2 class="subtitle is-3">
+            Advanced AI models in one-click
+          </h2>
+          <ul style="font-size: 1.5em;">
+            <li>- Integrate with Fiji, Ilastik, ImJoy</li>
+            <li>- Try model instantly with BioEngine</li>
+            <li>- Contribute your models via Github</li>
+            <li>- Link models to BioEngine Apps</li>
+          </ul>
+          <!-- <b-field label="Find a name">
```
Comment (MartinHjelmare) [inline src/views/Home.vue]: Should this be kept?

File: src/views/Home.vue [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+      console.log("Loading ImJoy...");
+      this.setupImJoyCore(repo_manifest.applications);
+      this.showModelInfo();
+    } catch (e) {
+      console.error(e);
+      alert(`Failed to fetch manifest file from the repo: ${e}.`);
+    }
+  },
+  computed: {
+    // filteredDataObj() {
```
Comment (MartinHjelmare) [inline src/views/Home.vue]: Commented code.

File: src/views/Home.vue [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+      const data = {
+        message: message.slice(0, 200),
+        onAction: function() {},
+        actionText: "Close",
+        duration: duration
+      };
+
+      this.$buefy.snackbar.open(data);
+    },
+    async setupImJoyCore(appSources) {
```
Comment (MartinHjelmare) [inline src/views/Home.vue]: Maybe break out the ImJoy code to its own module and just have a method here that calls an imported function or similar?

**Review by MartinHjelmare — COMMENTED** [review]
Nice!

File: src/views/Home.vue (hunk 3 above)
Comment (oeway) [reply src/views/Home.vue]: I agree, I actually did try that, but then find there are places in this function is updating the UI, it require more thinking on how to let it still change the UI.

**Comment by MartinHjelmare** [comment]
The buttons on the other pages, about, contribute etc are not working on mobile, neither Firefox or Chrome.

Should those pages not show as subfolders in the url?

**Comment by oeway** [comment]
> The buttons on the other pages, about, contribute etc are not working on mobile, neither Firefox or Chrome.
> 
> Should those pages not show as subfolders in the url?

8AD631DA-9DC1-4000-A965-78022D08B3CD

It looks fine on iOS, how does it looks on android?

**Comment by MartinHjelmare** [comment]
I mean the buttons on the actual about and contribute pages. Not the menu.

**Comment by MartinHjelmare** [comment]
This is how the about page looks for me on android Firefox. The x and the two slice hamburger menu buttons don't do anything when pushed. And there's no dedicated subdirectory for the page in the url.

Screenshot_20200622-182605

**Comment by oeway** [comment]
> This is how the about page looks for me on android Firefox. The x and the two slice hamburger menu buttons don't do anything when pushed. And there's no dedicated subdirectory for the page in the url.

I see, thanks for the report, I will fix that, I saw the same on ios.

**Comment by oeway** [comment]
@MartinHjelmare Could you test on you mobile with the latest version (reload a few times to make sure)?

**Comment by MartinHjelmare** [comment]
Now the x-button works. The maximize/minimize button is now active but doesn't make any change on my screen size. Maybe hide it on mobile screens?

**Comment by oeway** [comment]
> Now the x-button works. The maximize/minimize button is now active but doesn't make any change on my screen size. Maybe hide it on mobile screens?

Good point, just changed, please try it in the next preview.

**Pull request merged**